drfc_manager/pipelines/viewer.py

```
@@ -347,109 +435,106 @@ def start_streamlit_viewer(data: Dict[str, Any]) -> Dict[str, Any]:
-        stdout_file = open(streamlit_stdout_log, 'w')
-        stderr_file = open(streamlit_stderr_log, 'w')
         logger.info(f"Starting Streamlit viewer process: {' '.join(cmd)}")
-        process = subprocess.Popen(
-            cmd,
-            stdout=stdout_file,
-            stderr=stderr_file,
-            env=env
-        )
+        env = get_subprocess_env(env_vars)
+        env["DR_VIEWER_CONTAINERS"] = json.dumps(containers)
+        log_file = open(log_file_name, "w")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)